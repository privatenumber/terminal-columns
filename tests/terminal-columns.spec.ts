import { blue, bold, underline } from 'colorette';
/**
 * Dependencies are bundled in because they have a
 * higher Node.js requirement. Test compiled version
 * to verify it works with Node.js 12.
 */
import terminalColumns, { breakpoints } from '..';
// import terminalColumns, { breakpoints } from '../src';

const loremIpsumShort = 'Lorem ipsum dolor sit amet.';
const loremIpsumLong = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.';
const loremIpsumNewLines = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Dictumst quisque sagittis purus sit amet volutpat consequat mauris nunc.
Nunc sed augue lacus viverra vitae congue eu consequat ac.
Sit amet porttitor eget dolor morbi non arcu.
`.trim();

beforeAll(() => {
	process.stdout.columns = 100;
});

describe('edge cases', () => {
	describe('error handling', () => {
		test('missing columns', () => {
			expect(
				() => terminalColumns(
					[['']],
					[100, 200],
				),
			).toThrow('2 columns defined, but only 1 columns found');
		});

		test('invalid column', () => {
			expect(
				() => terminalColumns(
					[['']],
					['100'],
				),
			).toThrow('Invalid column width: "100"');
		});
	});

	describe('empty table', () => {
		test('no table', () => {
			// @ts-expect-error test error
			const table = terminalColumns();
			expect(table).toBe('');
		});

		test('no rows', () => {
			const table = terminalColumns([]);
			expect(table).toBe('');
		});

		test('no columns', () => {
			const table = terminalColumns([[], []]);
			expect(table).toBe('');
		});
	});

	test('inconsistent rows', () => {
		const table = terminalColumns([
			['A'],
			['B', 'B'],
			['C', 'C', 'C'],
		]);

		expect(table).toMatchSnapshot();
	});

	test('colored text', () => {
		const table = terminalColumns([
			[blue('A'.repeat(2))],
			['B', bold('B'.repeat(3))],
			['C', 'C', underline('C'.repeat(4))],
		]);

		expect(table).toMatchSnapshot();
	});
});

describe('padding', () => {
	test('overflowing padding reduction - even', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumShort,
				],
			],
			[
				{
					width: 1,
					paddingLeft: 200,
					paddingRight: 200,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('overflowing padding reduction - uneven', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumShort,
				],
			],
			[
				{
					width: 1,
					paddingLeft: 200,
					paddingRight: 100,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('overflowing content with overflowing padding reduction - even', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumLong,
				],
			],
			[
				{
					width: 'content-width',
					paddingLeft: 200,
					paddingRight: 200,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('align', () => {
	test('align right', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumNewLines,
				],
			],
			[
				{
					align: 'right',
				},
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('static widths', () => {
	test('fixed width', () => {
		const table = terminalColumns(
			[
				[loremIpsumShort, loremIpsumLong],
			],
			[10, 20],
		);

		expect(table).toMatchSnapshot();
	});

	test('overflowing width', () => {
		const table = terminalColumns(
			[
				[loremIpsumShort, loremIpsumLong],
			],
			[124, 152],
		);

		expect(table).toMatchSnapshot();
	});

	test('overflowing rows', () => {
		const table = terminalColumns(
			[
				[loremIpsumShort, loremIpsumShort],
				[loremIpsumShort, loremIpsumShort],
			],
			[10, 100],
		);

		expect(table).toMatchSnapshot();
	});

	test('overflowing width with padding', () => {
		const table = terminalColumns(
			[
				[loremIpsumShort, loremIpsumLong],
			],
			[
				{
					width: 124,
					paddingLeft: 6,
				},
				{
					width: 152,
					paddingLeft: 3,
					paddingRight: 6,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('percent widths', () => {
	test('50% 50%', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
			],
			[
				'50%',
				'50%',
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('50% 50% with padding', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
			],
			[
				{
					width: '50%',
					paddingLeft: 6,
					paddingRight: 4,
				},
				{
					width: '50%',
					paddingLeft: 3,
					paddingRight: 9,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('70% 30% with different content lengths', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
				[loremIpsumLong, loremIpsumShort],
			],
			['70%', '30%'],
		);

		expect(table).toMatchSnapshot();
	});

	test('100% 100% with padding', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
			],
			[
				{
					width: '100%',
					paddingLeft: 2,
					paddingRight: 2,
					paddingTop: 1,
				},
				{
					width: '100%',
					paddingLeft: 4,
					paddingRight: 4,
					paddingBottom: 1,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('content-width', () => {
	test('content-width with fixed width', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
				[loremIpsumLong, loremIpsumShort],
			],
			['content-width', 40],
		);

		expect(table).toMatchSnapshot();
	});

	test('content-width with padding', () => {
		const table = terminalColumns(
			[
				[loremIpsumLong, loremIpsumLong],
			],
			[
				{
					width: 'content-width',
					paddingLeft: 2,
					paddingRight: 1,
				},
				{
					width: 'content-width',
					paddingLeft: 1,
					paddingRight: 2,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('content-width with overflowing', () => {
		const table = terminalColumns(
			[
				[loremIpsumNewLines, loremIpsumNewLines, loremIpsumNewLines],
			],
			[
				{
					width: 'content-width',
					paddingLeft: 2,
					paddingRight: 1,
				},
				{
					width: 'content-width',
					paddingLeft: 1,
					paddingRight: 2,
				},
				{
					width: 'content-width',
					paddingTop: 1,
					paddingBottom: 1,
				},
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('auto', () => {
	test('event split', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
				],
			],
			[
				'auto',
				'auto',
				'auto',
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('event split - many', () => {
		const table = terminalColumns(
			[
				[
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
					loremIpsumShort,
				],
			],
			[
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
				'auto',
			],
		);

		expect(table).toMatchSnapshot();
	});

	test('mutli-row', () => {
		const table = terminalColumns(
			[
				[loremIpsumShort, loremIpsumNewLines, loremIpsumNewLines],
				[loremIpsumLong, loremIpsumLong, loremIpsumShort],
			],
		);

		expect(table).toMatchSnapshot();
	});
});

describe('breakpoints', () => {
	const getTable = () => terminalColumns(
		[
			[loremIpsumLong, loremIpsumLong],
			[loremIpsumLong, loremIpsumLong],
		],
		breakpoints({
			// Large screens
			'>= 90': ['content-width', 'auto'],

			// Normal screens
			'>= 25': ['100%', '100%'],

			'>= 0': {
				columns: ['content-width', 'content-width'],
				stdoutColumns: Number.POSITIVE_INFINITY,
			},
		}),
	);

	test('stdout: 25 - Too small', () => {
		process.stdout.columns = 25;
		const table = getTable();
		expect(table).toMatchSnapshot();
	});

	test('stdout: 90 - Normal', () => {
		process.stdout.columns = 90;
		const table = getTable();
		expect(table).toMatchSnapshot();
	});

	test('stdout: 150 - Very big', () => {
		process.stdout.columns = 150;
		const table = getTable();

		expect(table).toMatchSnapshot();
	});
});

describe('custom breakpoints function', () => {
	const getTable = () => terminalColumns(
		[
			[loremIpsumLong, loremIpsumLong],
			[loremIpsumLong, loremIpsumLong],
		],
		(stdoutColumns) => {
			// Large screens
			if (stdoutColumns > 90) {
				return ['content-width', 'auto'];
			}

			// Normal screens
			if (stdoutColumns > 25) {
				return ['100%', '100%'];
			}

			return {
				columns: ['content-width', 'content-width'],
				stdoutColumns: Number.POSITIVE_INFINITY,
			};
		},
	);

	test('stdout: 25 - Too small', () => {
		process.stdout.columns = 25;
		const table = getTable();
		expect(table).toMatchSnapshot();
	});

	test('stdout: 90 - Normal', () => {
		process.stdout.columns = 90;
		const table = getTable();
		expect(table).toMatchSnapshot();
	});

	test('stdout: 150 - Very big', () => {
		process.stdout.columns = 150;
		const table = getTable();

		expect(table).toMatchSnapshot();
	});
});
