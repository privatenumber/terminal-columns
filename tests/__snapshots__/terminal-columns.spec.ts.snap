// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align align right 1`] = `
" Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore
                                                                             et dolore magna aliqua.
                            Dictumst quisque sagittis purus sit amet volutpat consequat mauris nunc.
                                          Nunc sed augue lacus viverra vitae congue eu consequat ac.
                                                       Sit amet porttitor eget dolor morbi non arcu."
`;

exports[`auto event split - many 1`] = `
"Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   
sit amet.           sit amet.           sit amet.           sit amet.           sit amet.           
Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   Lorem ipsum dolor   
sit amet.           sit amet.           sit amet.           sit amet.           sit amet.           
Lorem ipsum dolor sit amet.                       Lorem ipsum dolor sit amet.                       "
`;

exports[`auto event split 1`] = `"Lorem ipsum dolor sit amet.      Lorem ipsum dolor sit amet.      Lorem ipsum dolor sit amet.       "`;

exports[`auto mutli-row 1`] = `
"Lorem ipsum dolor sit amet.      Lorem ipsum dolor sit amet,      Lorem ipsum dolor sit amet,       
                                 consectetur adipiscing elit, sed consectetur adipiscing elit, sed  
                                 do eiusmod tempor incididunt ut  do eiusmod tempor incididunt ut   
                                 labore et dolore magna aliqua.   labore et dolore magna aliqua.    
                                 Dictumst quisque sagittis purus  Dictumst quisque sagittis purus   
                                 sit amet volutpat consequat      sit amet volutpat consequat mauris
                                 mauris nunc.                     nunc.                             
                                 Nunc sed augue lacus viverra     Nunc sed augue lacus viverra vitae
                                 vitae congue eu consequat ac.    congue eu consequat ac.           
                                 Sit amet porttitor eget dolor    Sit amet porttitor eget dolor     
                                 morbi non arcu.                  morbi non arcu.                   
Lorem ipsum dolor sit amet,      Lorem ipsum dolor sit amet,      Lorem ipsum dolor sit amet.       
consectetur adipiscing elit, sed consectetur adipiscing elit, sed                                   
do eiusmod tempor incididunt ut  do eiusmod tempor incididunt ut                                    
labore et dolore magna aliqua.   labore et dolore magna aliqua.                                     "
`;

exports[`breakpoints stdout: 25 - Too small 1`] = `
"Lorem ipsum dolor sit    
amet, consectetur        
adipiscing elit, sed do  
eiusmod tempor incididunt
ut labore et dolore magna
aliqua.                  
Lorem ipsum dolor sit    
amet, consectetur        
adipiscing elit, sed do  
eiusmod tempor incididunt
ut labore et dolore magna
aliqua.                  
Lorem ipsum dolor sit    
amet, consectetur        
adipiscing elit, sed do  
eiusmod tempor incididunt
ut labore et dolore magna
aliqua.                  
Lorem ipsum dolor sit    
amet, consectetur        
adipiscing elit, sed do  
eiusmod tempor incididunt
ut labore et dolore magna
aliqua.                  "
`;

exports[`breakpoints stdout: 90 - Normal 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         "
`;

exports[`breakpoints stdout: 150 - Very big 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet,
                                                                                                                           consectetur adipiscing     
                                                                                                                           elit, sed do eiusmod tempor
                                                                                                                           incididunt ut labore et    
                                                                                                                           dolore magna aliqua.       
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet,
                                                                                                                           consectetur adipiscing     
                                                                                                                           elit, sed do eiusmod tempor
                                                                                                                           incididunt ut labore et    
                                                                                                                           dolore magna aliqua.       "
`;

exports[`content-width content-width with fixed width 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
et dolore magna aliqua.                                                                             
Lorem ipsum dolor sit amet, consectetur 
adipiscing elit, sed do eiusmod tempor  
incididunt ut labore et dolore magna    
aliqua.                                 
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
et dolore magna aliqua.                                                                             
Lorem ipsum dolor sit amet.             "
`;

exports[`content-width content-width with overflowing 1`] = `
"  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut      
  labore et dolore magna aliqua.                                                                    
  Dictumst quisque sagittis purus sit amet volutpat consequat mauris nunc.                          
  Nunc sed augue lacus viverra vitae congue eu consequat ac.                                        
  Sit amet porttitor eget dolor morbi non arcu.                                                     
 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut       
 labore et dolore magna aliqua.                                                                     
 Dictumst quisque sagittis purus sit amet volutpat consequat mauris nunc.                           
 Nunc sed augue lacus viverra vitae congue eu consequat ac.                                         
 Sit amet porttitor eget dolor morbi non arcu.                                                      
                                                                                                    
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
et dolore magna aliqua.                                                                             
Dictumst quisque sagittis purus sit amet volutpat consequat mauris nunc.                            
Nunc sed augue lacus viverra vitae congue eu consequat ac.                                          
Sit amet porttitor eget dolor morbi non arcu.                                                       
                                                                                                    "
`;

exports[`content-width content-width with padding 1`] = `
"  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut      
  labore et dolore magna aliqua.                                                                    
 Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut       
 labore et dolore magna aliqua.                                                                     "
`;

exports[`custom breakpoints function stdout: 25 - Too small 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
`;

exports[`custom breakpoints function stdout: 90 - Normal 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
ut labore et dolore magna aliqua.                                                         "
`;

exports[`custom breakpoints function stdout: 150 - Very big 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet,
                                                                                                                           consectetur adipiscing     
                                                                                                                           elit, sed do eiusmod tempor
                                                                                                                           incididunt ut labore et    
                                                                                                                           dolore magna aliqua.       
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.Lorem ipsum dolor sit amet,
                                                                                                                           consectetur adipiscing     
                                                                                                                           elit, sed do eiusmod tempor
                                                                                                                           incididunt ut labore et    
                                                                                                                           dolore magna aliqua.       "
`;

exports[`edge cases colored text 1`] = `
"[34mAA[39m                                                                                                  
B                               [1mBBB[22m                                                                 
C                               C                                [4mCCCC[24m                               "
`;

exports[`edge cases inconsistent rows 1`] = `
"A                                                                                                   
B                                B                                                                  
C                                C                                C                                 "
`;

exports[`edge cases infinite width 1`] = `
"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC"
`;

exports[`padding overflowing content with overflowing padding reduction - even 1`] = `
"                                                 L                                                  
                                                 o                                                  
                                                 r                                                  
                                                 e                                                  
                                                 m                                                  
                                                 i                                                  
                                                 p                                                  
                                                 s                                                  
                                                 u                                                  
                                                 m                                                  
                                                 d                                                  
                                                 o                                                  
                                                 l                                                  
                                                 o                                                  
                                                 r                                                  
                                                 s                                                  
                                                 i                                                  
                                                 t                                                  
                                                 a                                                  
                                                 m                                                  
                                                 e                                                  
                                                 t                                                  
                                                 ,                                                  
                                                 c                                                  
                                                 o                                                  
                                                 n                                                  
                                                 s                                                  
                                                 e                                                  
                                                 c                                                  
                                                 t                                                  
                                                 e                                                  
                                                 t                                                  
                                                 u                                                  
                                                 r                                                  
                                                 a                                                  
                                                 d                                                  
                                                 i                                                  
                                                 p                                                  
                                                 i                                                  
                                                 s                                                  
                                                 c                                                  
                                                 i                                                  
                                                 n                                                  
                                                 g                                                  
                                                 e                                                  
                                                 l                                                  
                                                 i                                                  
                                                 t                                                  
                                                 ,                                                  
                                                 s                                                  
                                                 e                                                  
                                                 d                                                  
                                                 d                                                  
                                                 o                                                  
                                                 e                                                  
                                                 i                                                  
                                                 u                                                  
                                                 s                                                  
                                                 m                                                  
                                                 o                                                  
                                                 d                                                  
                                                 t                                                  
                                                 e                                                  
                                                 m                                                  
                                                 p                                                  
                                                 o                                                  
                                                 r                                                  
                                                 i                                                  
                                                 n                                                  
                                                 c                                                  
                                                 i                                                  
                                                 d                                                  
                                                 i                                                  
                                                 d                                                  
                                                 u                                                  
                                                 n                                                  
                                                 t                                                  
                                                 u                                                  
                                                 t                                                  
                                                 l                                                  
                                                 a                                                  
                                                 b                                                  
                                                 o                                                  
                                                 r                                                  
                                                 e                                                  
                                                 e                                                  
                                                 t                                                  
                                                 d                                                  
                                                 o                                                  
                                                 l                                                  
                                                 o                                                  
                                                 r                                                  
                                                 e                                                  
                                                 m                                                  
                                                 a                                                  
                                                 g                                                  
                                                 n                                                  
                                                 a                                                  
                                                 a                                                  
                                                 l                                                  
                                                 i                                                  
                                                 q                                                  
                                                 u                                                  
                                                 a                                                  
                                                 .                                                  "
`;

exports[`padding overflowing padding reduction - even 1`] = `
"                                                 L                                                  
                                                 o                                                  
                                                 r                                                  
                                                 e                                                  
                                                 m                                                  
                                                 i                                                  
                                                 p                                                  
                                                 s                                                  
                                                 u                                                  
                                                 m                                                  
                                                 d                                                  
                                                 o                                                  
                                                 l                                                  
                                                 o                                                  
                                                 r                                                  
                                                 s                                                  
                                                 i                                                  
                                                 t                                                  
                                                 a                                                  
                                                 m                                                  
                                                 e                                                  
                                                 t                                                  
                                                 .                                                  "
`;

exports[`padding overflowing padding reduction - uneven 1`] = `
"                                                                  L                                 
                                                                  o                                 
                                                                  r                                 
                                                                  e                                 
                                                                  m                                 
                                                                  i                                 
                                                                  p                                 
                                                                  s                                 
                                                                  u                                 
                                                                  m                                 
                                                                  d                                 
                                                                  o                                 
                                                                  l                                 
                                                                  o                                 
                                                                  r                                 
                                                                  s                                 
                                                                  i                                 
                                                                  t                                 
                                                                  a                                 
                                                                  m                                 
                                                                  e                                 
                                                                  t                                 
                                                                  .                                 "
`;

exports[`percent widths 50% 50% 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscingLorem ipsum dolor sit amet, consectetur adipiscing
elit, sed do eiusmod tempor incididunt ut labore  elit, sed do eiusmod tempor incididunt ut labore  
et dolore magna aliqua.                           et dolore magna aliqua.                           "
`;

exports[`percent widths 50% 50% with padding 1`] = `
"      Lorem ipsum dolor sit amet, consectetur        Lorem ipsum dolor sit amet,                    
      adipiscing elit, sed do eiusmod tempor         consectetur adipiscing elit, sed do            
      incididunt ut labore et dolore magna           eiusmod tempor incididunt ut labore et         
      aliqua.                                        dolore magna aliqua.                           "
`;

exports[`percent widths 70% 30% with different content lengths 1`] = `
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do       Lorem ipsum dolor sit amet,   
eiusmod tempor incididunt ut labore et dolore magna aliqua.           consectetur adipiscing elit,  
                                                                      sed do eiusmod tempor         
                                                                      incididunt ut labore et dolore
                                                                      magna aliqua.                 
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do       Lorem ipsum dolor sit amet.   
eiusmod tempor incididunt ut labore et dolore magna aliqua.                                         "
`;

exports[`percent widths 100% 100% with padding 1`] = `
"                                                                                                    
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut      
  labore et dolore magna aliqua.                                                                    
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut    
    labore et dolore magna aliqua.                                                                  
                                                                                                    "
`;

exports[`process postprocess 1`] = `
"LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNT UT LABORE 
et dolore magna aliqua.                                                                             
DICTUMST QUISQUE SAGITTIS PURUS SIT AMET VOLUTPAT CONSEQUAT MAURIS NUNC.                            
nunc sed augue lacus viverra vitae congue eu consequat ac.                                          
SIT AMET PORTTITOR EGET DOLOR MORBI NON ARCU.                                                       "
`;

exports[`process postprocess ignores vertical padding 1`] = `
"                                                  Lorem ipsum dolor sit amet, consectetur adipiscing
postprocessed                                     elit, sed do eiusmod tempor incididunt ut labore  
postprocessed                                     et dolore magna aliqua.                           
postprocessed                                     Dictumst quisque sagittis purus sit amet volutpat 
postprocessed                                     consequat mauris nunc.                            
postprocessed                                     Nunc sed augue lacus viverra vitae congue eu      
postprocessed                                     consequat ac.                                     
postprocessed                                     Sit amet porttitor eget dolor morbi non arcu.     
postprocessed                                                                                       
                                                                                                    
                                                                                                    
                                                                                                    "
`;

exports[`process preprocess 1`] = `
"LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNT UT LABORE 
ET DOLORE MAGNA ALIQUA.                                                                             
DICTUMST QUISQUE SAGITTIS PURUS SIT AMET VOLUTPAT CONSEQUAT MAURIS NUNC.                            
NUNC SED AUGUE LACUS VIVERRA VITAE CONGUE EU CONSEQUAT AC.                                          
SIT AMET PORTTITOR EGET DOLOR MORBI NON ARCU.                                                       "
`;

exports[`static widths fixed width 1`] = `
"Lorem     Lorem ipsum dolor   
ipsum     sit amet,           
dolor sit consectetur         
amet.     adipiscing elit, sed
          do eiusmod tempor   
          incididunt ut labore
          et dolore magna     
          aliqua.             "
`;

exports[`static widths overflowing rows 1`] = `
"Lorem     
ipsum     
dolor sit 
amet.     
Lorem ipsum dolor sit amet.                                                                         
Lorem     
ipsum     
dolor sit 
amet.     
Lorem ipsum dolor sit amet.                                                                         "
`;

exports[`static widths overflowing width 1`] = `
"Lorem ipsum dolor sit amet.                                                                         
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
et dolore magna aliqua.                                                                             "
`;

exports[`static widths overflowing width with padding 1`] = `
"      Lorem ipsum dolor sit amet.                                                                   
   Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt        
   ut labore et dolore magna aliqua.                                                                "
`;
